compiler-rt/lib/builtins/aarch64/sme-libc-routines.c

```diff
@@ -22,3 +23,46 @@ const void *__arm_sc_memchr(const void *src, int c,
 
   return NULL;
 }
+
+#ifndef __ARM_FEATURE_UNALIGNED
```

Comment:
To avoid creating many different variants and various ifdefs, does it make sense to choose the implementation (either `sme-libc-mem-routines.S` (optimised) or `sme-libc-routines.c` (basic)) in CMakeLists.txt ? That way we can also let cmake emit a warning that the unoptimised SME routines are chosen, because of the constraints given by the compilation flags to compiler-rt.

Reply:
The issue is it isn't either this or that. For example, for AArch64 soft-nofp with unaligned accesses, some of the functions will come from the optimised .S file, and others from the basic .c file.

In order to have it work without the #ifdefs and avoid multiple definition errors, I will have to refactor the files into smaller ones and select them accordingly in the CMakeLists.txt. I am having a go at this option.